server/src/main/java/org/elasticsearch/indices/CachedTimestampFieldInfo.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices;
+
+import org.elasticsearch.index.mapper.DateFieldMapper;
+import org.elasticsearch.index.shard.IndexLongFieldRange;
+
+/**
+ * Data holder of timestamp fields held in cluster state IndexMetadata.
+ */
+public class CachedTimestampFieldInfo {
```

Comment:
make it final? Perhaps I would call it `DateFieldRange`. Did you think it's important to include "Cached" in the name? Also Timestamp has a reference to `@timestamp` that maybe we should loosen up? (not a big deal, curious to know what you think)

Reply:
Done. Changed class name to `DateFieldRangeInfo`. I didn't want to use just `DateFieldRange` since that implies a range only and this object is used for the field types as well than the ranges. But happy to change it if you don't like the new name.

Reply:
it's fine!